WeatherApp/ViewControllers/WeatherViewController.swift

```diff
@@ -9,27 +9,67 @@
 import UIKit
 
 class WeatherViewController: UIViewController {
-
     let tableView = UITableView()
+    let weatherService = WeatherApiService()
+    let cities = ["Zagreb", "Split", "Osijek", "Rijeka"]
+    var weatherData: [WeatherData] = [] {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
     
-    override func loadView() {
-        super.loadView()
-        
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.dataSource = self
```

Comment:
You can extract these `tableView` lines into separate `func` or include them into `setupUI`

Reply:
@termodinamika